Makefile

```diff
@@ -157,9 +146,10 @@ build/plugin/%.o: plugin/%.cc
 	$(CXX) -std=c++0x -c $(CFLAGS) -c $< -o $@
 
 # A nvcc bug cause this to generate "generic/xxx.h" dependencies from torch headers.
-# $(NVCC) $(NVCCFLAGS) -Xcompiler "$(CFLAGS)" -M -MT build/plugin/$*_gpu.o $< >build/plugin/$*_gpu.d
+# Use CXX to generate dependency instead.
 build/plugin/%_gpu.o: plugin/%.cu
 	@mkdir -p $(@D)
+	$(CXX) -std=c++0x $(CFLAGS) -MM -MT build/plugin/$*_gpu.o $< >build/plugin/$*_gpu.d
```

Comment:
nvcc's dependency generation doesn't work with torch headers. I used CXX to generate dependency for .cu files instead. It seems to work just fine.